SYCL/ESIMD/api/functional/functions/functions_1d_select.hpp

```diff
@@ -0,0 +1,281 @@
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file provides functions for tests on simd 1d select function.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "../shared_element.hpp"
+#include "common.hpp"
+
+// for std::ceil
+#include <cmath>
+// for std::numeric_limits
+#include <limits>
+
+namespace esimd = sycl::ext::intel::experimental::esimd;
+
+namespace esimd_test::api::functional::functions {
+
+namespace details {
+
+constexpr int ceil(float arg) {
+  return (arg > static_cast<int>(arg)) ? arg + 1 : arg;
+}
+
+} // namespace details
+
+// Aliases to provide size or stride values to test.
+// This is the syntax sugar and they have the same type.
+template <int N> using stride_type = std::integral_constant<int, N>;
+template <int N> using size_type = std::integral_constant<int, N>;
+template <int N> using offset_type = std::integral_constant<int, N>;
+
+using use_offset = std::true_type;
+using do_not_use_offset = std::true_type;
+
+// Descriptor class for the case of calling simd<T,N>::select function.
+struct select_rval {
+  static std::string get_description() { return "select rvalue"; }
+
+  template <typename DataT, int NumElems, int NumSelectedElems, int Stride>
+  static bool call_operator(const DataT *const ref_1, const DataT *const ref_2,
+                            DataT *const out, size_t offset) {
+    esimd::simd<DataT, NumElems> simd_obj;
+    simd_obj.copy_from(ref_1);
+    auto select_result =
+        simd_obj.template select<NumSelectedElems, Stride>(offset);
+
+    for (size_t i = 0; i < NumSelectedElems; ++i) {
+      select_result[i] = ref_2[i];
+    }
+    simd_obj.copy_to(out);
+
+    return std::is_same_v<
+        decltype(select_result),
+        simd_view<esimd::simd<DataT, NumElems>,
+                  region1d_t<DataT, NumSelectedElems, Stride>>>;
+  }
+};
+
+// Descriptor class for the case of calling simd<T,N>::select function.
+struct select_lval {
+  static std::string get_description() { return "select lvalue"; }
+
+  template <typename DataT, int NumElems, int NumSelectedElems, int Stride>
+  static bool call_operator(const DataT *const ref_1, const DataT *const ref_2,
+                            DataT *const out, size_t offset) {
+    simd<DataT, NumElems> src_simd_obj;
+    src_simd_obj.copy_from(ref_1);
+
+    simd<DataT, NumSelectedElems> dst_simd_obj;
+    dst_simd_obj.copy_from(ref_2);
+
+    src_simd_obj.template select<NumSelectedElems, Stride>(offset) =
+        dst_simd_obj;
+    src_simd_obj.copy_to(out);
+
+    return true;
+  }
+};
+
+// Descriptor class for the case of calling simd<T,N>::select function.
+struct select_simd_view {
```

Comment:
```suggestion
struct select_simd_view_rval {
```
Given the naming above. What about l-value?

Reply:
This comment has been applied [here](https://github.com/intel/llvm-test-suite/pull/875/commits/03d94c35b4a95e3ce26c99e1197bc50ccdc652aa#diff-295141c99d991cbb4971abc3bd754b2f17400cade3b579ac7bc36b28a1450dc3R92)